docs/pages/debugging/devtools-plugins.mdx

```diff
@@ -0,0 +1,201 @@
+
+```jsx
+if (__DEV__) {
+  // initialize your DevTools plugin here
+}
+```
+
+In some cases, you may need to interact with a plugin directly. All plugins communicate through exports from `expo/devtools`, and you can send and listen to messages through `connectPluginFromDevToolsAsync`:
+
+```jsx
+import { connectPluginFromDevToolsAsync } from 'expo/devtools';
+
+const client = await connectPluginFromDevToolsAsync();
+
+client.sendMessage('ping', { from: 'app' });
+client.addMessageListener('pong', (data) => {
+  console.log('receiving pong', data);
+});
+```
+
+### Compatibility with Expo Go and Development Builds
+
+Dev tools plugins themselves should only include JavaScript code, meaning they are generally compatible with both Expo Go and [Development Builds](development-builds/introduction/), and it should not require creating a new development build in order to add a dev tools plugin. If the underlying module that the plugin is inspecting is from a package that includes native code that is not already part of Expo Go, then you would need to create a new development build with that package included in order to use both the component and the plugin.
+
+For example, if you had a dev tools plugin that inspected [React Native Firebase](), that would not work with Expo Go, as React Native Firebase includes native code that is not already part of Expo Go. You would need to create a development build in order to use React Native Firebase, and then you could also use the dev tools plugin.
+
+## Using a dev tools plugin
+
+Once your dev tools plugin is installed and any required connecting code is added to your project. Start your dev server up with `npx expo start`, and then press <kbd>shift</kbd> + <kbd>m</kbd> to open the list of dev tools plugins. Select the plugin you want to use, and it will open in a new Chrome window.
+
+> When starting the a dev server with the Expo CLI, there is an option to press <kbd>?</kbd> to "show all commands". This will show additional commands, including the shortcut to open "more tools." Dev tools plugins can also be selected in this menu.
+
+## Expo dev tools Plugins
+
+Expo provides some dev tools plugins for common debugging tasks. Follow the directions below to start using them in your app.
+
+### React Navigation
+
+Inspired by [@react-navigation/devtools](https://github.com/react-navigation/react-navigation/tree/4797ace/packages/devtools) and [flipper-plugin-react-navigation](https://github.com/react-navigation/react-navigation/tree/4797ace/packages/flipper-plugin-react-navigation), the React Navigation Dev Tools Plugin allows you to view a history of [React Navigation](https://reactnavigation.org/) actions and state. Rewind to previous points in your navigation history and send deep links to your app. Since Expo Router is built upon React Navigation, this plugin is fully-compatible with [Expo Router](/router/introduction).
```

Comment:
```suggestion
Inspired by [`@react-navigation/devtools`](https://github.com/react-navigation/react-navigation/tree/4797ace/packages/devtools) and [`flipper-plugin-react-navigation`](https://github.com/react-navigation/react-navigation/tree/4797ace/packages/flipper-plugin-react-navigation), the React Navigation dev tools plugin allows seeing history of [React Navigation](https://reactnavigation.org/) actions and state. You can also rewind to previous points in your navigation history and send deep links to your app. Since Expo Router is built upon React Navigation, this plugin is fully compatible with [Expo Router](/router/introduction).
```

- Since this is a package, let's use inline code syntax highlights